influxql/select.go

```diff
@@ -850,13 +857,11 @@ func (b *exprIteratorBuilder) buildCallIterator(expr *Call) (Iterator, error) {
 			}
 			return itr, nil
 		case "median":
-			opt := b.opt
-			opt.Ordered = true
-			input, err := buildExprIterator(expr.Args[0].(*VarRef), b.ic, b.sources, opt, false)
+			input, err := buildOrderedExprIterator(expr.Args[0].(*VarRef), b.ic, b.sources, b.opt, false)
 			if err != nil {
 				return nil, err
 			}
-			return newMedianIterator(input, opt)
+			return newMedianIterator(input, b.opt)
```

Comment:
Before this change, here in `median` and in `percentile`, `opt.Ordered` was set to true when passed to `newMedianIterator` or `newPercentileIterator`.

But now, `b.opt` is passed by value to `buildOrderedExprIterator`, meaning `b.opt.Ordered` is not necessarily true.

Is that okay here? Should the signature of `buildOrderedExprIterator` change to accept a reference to `IteratorOptions` rather than the value?